### PR TITLE
NodeJS Homework - Week 2

### PR DESCRIPTION
**Pull request by Yasser2H**

### CODE REVIEW

File: nodejs/week2/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    if (isNaN(reqParams.x)) {
+        return res.status(406).json({error: "The first input is not a number"});
+    } else if (isNaN(reqParams.y)) {
+        return res.status(406).json({error: "The second input is not a number"});
+    } else {
+        const firstNum = parseInt(reqParams.x);
+        const secondNum = parseInt(reqParams.y);
+        const result = firstNum * secondNum;
+        res.send("First Number: " + firstNum + " Second Number: " + secondNum + " Multiplication: " + result);
+    }
```
Comment (Laila1222) [inline nodejs/week2/warmup/index.js:32]: Good

File: nodejs/week2/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    const reqQuery = req.query;
+    if (isNaN(reqQuery.first)) {
+        return res.status(406).json({error: "The first input is not a number"});
+    } else if (isNaN(reqQuery.second)) {
+        return res.status(406).json({error: "The second input is not a number"});
+    } else {
+        const firstNum = parseInt(reqQuery.first);
+        const secondNum = parseInt(reqQuery.second);
+        const result = firstNum + secondNum;
+        res.send("First Number: " + firstNum + " Second Number: " + secondNum + " Sum: " + result);
```
Comment (Laila1222) [inline nodejs/week2/warmup/index.js:17]: Good job

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            } else {
+                response.send(meals.slice(0, reqQuery.limit));
+            }
+        } else {
+            response.send(meals);
+        }
+    } catch (error) {
+        throw response.send(400).json(error);
+    }
+});
```
Comment (Laila1222) [inline nodejs/week2/meal-sharing/src/backend/api/meals-router.js:36]: Good job! Route works well with the different parameters.

File: nodejs/week2/meal-sharing/src/backend/api/reservations-router.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        if (filteredReservations.length === 0){
+            return response.status(200).json({message: "Reservation not found!"});
+        } else {
+            response.send(filteredReservations[0]);
+        }
+
+    } catch (error) {
+        throw response.send(400).json(error);
+    }
```
Comment (Laila1222) [inline nodejs/week2/meal-sharing/src/backend/api/reservations-router.js:28]: Good!

**Review by Laila1222 — APPROVED** [review]
Excellent job!

**Comment by Yasser2H** [comment]
Thank you, I'm glad there are no issues :)

**Pull request merged**